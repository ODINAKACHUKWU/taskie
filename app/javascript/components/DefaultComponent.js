import React from "react";

function DefaultComponent() {
  return <h1 className="welcome-text">Welcome to the Taskie app!</h1>;
}

export default DefaultComponent;
