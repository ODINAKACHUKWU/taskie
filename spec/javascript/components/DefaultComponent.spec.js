import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import DefaultComponent from "../../../app/javascript/components/DefaultComponent";

describe("rendering DefaultComponent component", () => {
  it("renders DefaultComponent component without crashing", () => {
    shallow(<DefaultComponent />);
  });

  it("renders header content", () => {
    const wrapper = shallow(<DefaultComponent />);
    const content = wrapper.find("h1").text();

    expect(content).toEqual("Welcome to the Taskie app!");
  });
});

describe("snapshot", () => {
  it("DefaultComponent snapshot", () => {
    const tree = shallow(<DefaultComponent />);

    expect(toJSON(tree)).toMatchSnapshot();
  });
});
