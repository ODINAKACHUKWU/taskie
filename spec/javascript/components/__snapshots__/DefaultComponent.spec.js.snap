// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot DefaultComponent snapshot 1`] = `
<h1
  className="welcome-text"
>
  Welcome to the Taskie app!
</h1>
`;
